### ISSUE TITLE
Failing to compile after adding plugin

### ISSUE THREAD

**Issue by newtmex** [mapsplugin/cordova-plugin-googlemaps#2492]
```:CordovaLib:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:compileDebugAidl UP-TO-DATE
:CordovaLib:compileDebugRenderscript UP-TO-DATE
:CordovaLib:checkDebugManifest UP-TO-DATE
:CordovaLib:generateDebugBuildConfig UP-TO-DATE
:CordovaLib:prepareLintJar UP-TO-DATE
:CordovaLib:generateDebugResValues UP-TO-DATE
:CordovaLib:generateDebugResources UP-TO-DATE
:CordovaLib:packageDebugResources UP-TO-DATE
:CordovaLib:platformAttrExtractor UP-TO-DATE
:CordovaLib:processDebugManifest UP-TO-DATE
:CordovaLib:processDebugResources UP-TO-DATE
:CordovaLib:generateDebugSources UP-TO-DATE
:CordovaLib:javaPreCompileDebug UP-TO-DATE
:CordovaLib:compileDebugJavaWithJavac UP-TO-DATE
:CordovaLib:processDebugJavaRes NO-SOURCE
:CordovaLib:transformClassesAndResourcesWithPrepareIntermediateJarsForDebug UP-TO-DATE
:app:preBuild UP-TO-DATE
:app:preDebugBuild FAILED```

Above is part of what I get when I run ```ionic cordova run android``` after installing the plugin

Any help??

**Comment by wf9a5m75** [comment]
Log message does not include any information to detect the reason.

**Comment by newtmex** [comment]
Okay, this is the full log message
`✔ Running command - done!
Android Studio project detected
ANDROID_HOME=/Users/<user>/Library/Android/sdk
JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home
studio
Subproject Path: CordovaLib
Subproject Path: app
publishNonDefault is deprecated and has no effect anymore. All variants are now published.
Configuration 'compile' in project ':app' is deprecated. Use 'implementation' instead.
The Task.leftShift(Closure) method has been deprecated and is scheduled tobe removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_ehwfs7vmmwg6hvy7hi0hhlvng.run(/Users/<user>/<dir>/<dir>/<dir>/< app dir>/platforms/android/app/build.gradle:144)
:CordovaLib:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:compileDebugAidl UP-TO-DATE
:CordovaLib:compileDebugRenderscript UP-TO-DATE
:CordovaLib:checkDebugManifest UP-TO-DATE
:CordovaLib:generateDebugBuildConfig UP-TO-DATE
:CordovaLib:prepareLintJar UP-TO-DATE
:CordovaLib:generateDebugResValues UP-TO-DATE
:CordovaLib:generateDebugResources UP-TO-DATE
:CordovaLib:packageDebugResources UP-TO-DATE
:CordovaLib:platformAttrExtractor UP-TO-DATE
:CordovaLib:processDebugManifest UP-TO-DATE
:CordovaLib:processDebugResources UP-TO-DATE
:CordovaLib:generateDebugSources UP-TO-DATE
:CordovaLib:javaPreCompileDebug UP-TO-DATE
:CordovaLib:compileDebugJavaWithJavac UP-TO-DATE
:CordovaLib:processDebugJavaRes NO-SOURCE
:CordovaLib:transformClassesAndResourcesWithPrepareIntermediateJarsForDebug UP-TO-DATE
:app:preBuild UP-TO-DATE
:app:preDebugBuild FAILED
14 actionable tasks: 1 executed, 13 up-to-date`

I am using Ionic

When I run `ionic info` below is my environment info

`
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.9.2
    ionic (Ionic CLI) : 3.9.2

global packages:

    Cordova CLI : 8.0.0

local packages:

    @ionic/app-scripts : 3.1.0
    Cordova Platforms  : android 7.0.0 browser 5.0.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    Node : v9.2.1
    npm  : 6.1.0
    OS   : macOS High Sierra`

**Comment by wf9a5m75** [comment]
Unfortunately there is no information in the log. Please share your project files on GitHub.

**Comment by newtmex** [comment]
> Unfortunately there is no information in the log. Please share your project files on GitHub.

I sent you an invite to the private repo. Thanks for your assistance.

**Comment by wf9a5m75** [comment]
Thank you for sharing your project.

The error message obviously describes why the building failed.

```
:app:compileDebugAidl
:app:compileDebugRenderscript
:app:checkDebugManifest
:app:generateDebugBuildConfig
:app:generateDebugResources
:app:mergeDebugResources
/Users/masashi/Documents/workspace2/test/urbanmedics/platforms/android/app/src/main/AndroidManifest.xml:18:5-74 Error:
	uses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [:tbxml-android:] /Users/masashi/.gradle/caches/transforms-1/files-1.1/tbxml-android.aar/e0cfe58a45cbe1c182e4f0ac817d9c1d/AndroidManifest.xml as the library might be using APIs not available in 16
	Suggestion: use a compatible library with a minSdk of at most 16,
		or increase this project's minSdk version to at least 19,
		or use tools:overrideLibrary="za.co.twyst" to force usage (may lead to runtime failures)
:app:processDebugManifest

:app:processDebugManifest FAILED

FAILURE: Build failed with an exception.

```

You didn't include the message, but if you include this message, I don't need to spend my time.
You should not omit log message.

**Comment by newtmex** [comment]
Thanks for the response and assistance. But the log message I am getting is totally different. I even changed my `<uses-sdk android:minSdkVersion="16" android:targetSdkVersion="26" />` to `<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="26" />` but I still get this log message after running `ionic cordova build android`

```
[INFO] Running app-scripts build: --platform android --target cordova

[19:07:04]  build dev started ...
[19:07:04]  clean started ...
[19:07:04]  clean finished in 10 ms
[19:07:04]  copy started ...
[19:07:04]  deeplinks started ...
[19:07:04]  deeplinks finished in 113 ms
[19:07:04]  transpile started ...
[19:07:09]  transpile finished in 4.82 s
[19:07:09]  preprocess started ...
[19:07:09]  preprocess finished in 1 ms
[19:07:09]  webpack started ...
[19:07:09]  copy finished in 5.12 s
[19:07:16]  webpack finished in 6.68 s
[19:07:16]  sass started ...
Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.
[19:07:17]  sass finished in 1.21 s
[19:07:17]  postprocess started ...
[19:07:17]  postprocess finished in 26 ms
[19:07:17]  lint started ...
[19:07:17]  build dev finished in 13.00 s
> ionic cordova prepare
> cordova prepare
✔ Running command - done!
Android Studio project detected
> cordova build android
⠦ Running command [19:07:25]  tslint: src/pages/direction/direction.ts, lines: 3 - 5
            All imports on this line are unused.

       L2:  import { IonicPage, NavController, NavParams } from 'ionic-angular';
       L3:  import {
       L4:    MarkerTypeId, IMapOptions, IBox, IMarkerIconInfo
       L5:  } from 'angular-maps';
⠧ Running command        L6:  import { GoogleMap, GoogleMaps, Environment } from '@ionic-native/google-maps';

[19:07:25]  tslint: src/pages/direction/direction.module.ts, line: 5
            'MarkerTypeId' is declared but never used.

       L5:  import {MapModule, MapAPILoader, MarkerTypeId, IMapOptions, IBox, IMarkerIconInfo, WindowRef,
       L6:    DocumentRef, MapServiceFactory,

[19:07:25]  tslint: src/pages/direction/direction.module.ts, line: 5
            'IMapOptions' is declared but never used.

       L5:  import {MapModule, MapAPILoader, MarkerTypeId, IMapOptions, IBox, IMarkerIconInfo, WindowRef,
       L6:    DocumentRef, MapServiceFactory,

[19:07:25]  tslint: src/pages/direction/direction.module.ts, line: 5
            'IBox' is declared but never used.

       L5:  import {MapModule, MapAPILoader, MarkerTypeId, IMapOptions, IBox, IMarkerIconInfo, WindowRef,
       L6:    DocumentRef, MapServiceFactory,

[19:07:25]  tslint: src/pages/direction/direction.module.ts, line: 5
            'IMarkerIconInfo' is declared but never used.

       L5:  import {MapModule, MapAPILoader, MarkerTypeId, IMapOptions, IBox, IMarkerIconInfo, WindowRef,
       L6:    DocumentRef, MapServiceFactory,

[19:07:25]  tslint: src/pages/direction/direction.module.ts, line: 6
            'MapServiceFactory' is declared but never used.

       L5:  import {MapModule, MapAPILoader, MarkerTypeId, IMapOptions, IBox, IMarkerIconInfo, WindowRef,
       L6:    DocumentRef, MapServiceFactory,
       L7:    BingMapAPILoaderConfig, BingMapAPILoader,

[19:07:25]  lint finished in 7.45 s
✔ Running command - done!
Android Studio project detected
ANDROID_HOME=/Users/levichimezie/Library/Android/sdk
JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home
studio
Subproject Path: CordovaLib
Subproject Path: app
publishNonDefault is deprecated and has no effect anymore. All variants are now published.
Configuration 'compile' in project ':app' is deprecated. Use 'implementation' instead.
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action)instead.
        at build_ehwfs7vmmwg6hvy7hi0hhlvng.run(/Users/levichimezie/js_htdocs/PROJECTS/UrbanMedics/app/platforms/android/app/build.gradle:144)
:CordovaLib:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:compileDebugAidl UP-TO-DATE
:CordovaLib:compileDebugRenderscript UP-TO-DATE
:CordovaLib:checkDebugManifest UP-TO-DATE
:CordovaLib:generateDebugBuildConfig UP-TO-DATE
:CordovaLib:prepareLintJar UP-TO-DATE
:CordovaLib:generateDebugResValues UP-TO-DATE
:CordovaLib:generateDebugResources UP-TO-DATE
:CordovaLib:packageDebugResources UP-TO-DATE
:CordovaLib:platformAttrExtractor UP-TO-DATE
:CordovaLib:processDebugManifest UP-TO-DATE
:CordovaLib:processDebugResources UP-TO-DATE
:CordovaLib:generateDebugSources UP-TO-DATE
:CordovaLib:javaPreCompileDebug UP-TO-DATE
:CordovaLib:compileDebugJavaWithJavac UP-TO-DATE
:CordovaLib:processDebugJavaRes NO-SOURCE
:CordovaLib:transformClassesAndResourcesWithPrepareIntermediateJarsForDebug UP-TO-DATE
:app:preBuild UP-TO-DATE
:app:preDebugBuild FAILED
14 actionable tasks: 1 executed, 13 up-to-date
```

Maybe I am not doing something right.

**Comment by wf9a5m75** [comment]
Please take screen video, and share it on Youtube.

**Comment by wf9a5m75** [comment]
Thank you. I understand your situation. But I wonder why you can not get any error messages.

**Comment by wf9a5m75** [comment]
Please try https://build.phonegap.com/ 

You can build one private repo as free.

**Comment by newtmex** [comment]
HI @wf9a5m75 , can you please change the `minSdkVersion` in your config.xml and try to build the app to see if it will build successfully. Thanks.

**Comment by wf9a5m75** [comment]
I'm not free Q&A service.

**Comment by wf9a5m75** [comment]
You should search on Google, or the [document repository](https://github.com/mapsplugin/cordova-plugin-googlemaps-doc).

**Comment by newtmex** [comment]
Okay, thanks. I solved the issue by updating my Cordova to 8.1.2 `(cordova-lib@8.1.1)` using `npm update cordova`, then useful error messages came up and I resolved them accordingly. Thanks for the time and attention you gave to me.
